geonode/base/api/permissions.py

```diff
@@ -136,6 +136,26 @@ def has_object_permission(self, request, view, obj):
         return IsOwnerOrAdmin.has_object_permission(self, request, view, obj)
 
 
+class IsAdminOrReadOnly(permissions.BasePermission):
```

Comment:
Using `IsSelfOrAdminOrReadOnly` creates a security breach because it would be possible for users to edit objects if the object `id` is the same of the user id, for some coincidence.
https://github.com/GeoNode/geonode/blob/master/geonode/base/api/permissions.py#L46

Reply:
Rather then creating additional combinations IMHO the IsSelf method should be fixed to enforce a check on the obj model before doing the comparison.

Reply:
Updated ✅